### PR TITLE
isolate tests in child processes

### PR DESCRIPTION
**Pull request by Peyman-EsteghamatRad**
With this PR testcases are isolated in single processes. This allows to keep executing all testcases even if some tests fail due to an unhandled signal (e.g. SIGSEGV).

### CODE REVIEW

File: utest.h [hunk 1]
```diff
@@ -33,6 +33,10 @@
 #ifndef SHEREDOM_UTEST_H_INCLUDED
 #define SHEREDOM_UTEST_H_INCLUDED
 
+#include <cstdlib>
```
Comment (sheredom) [inline utest.h:36]: You are gonna run into some nasty cross platform issues here I'd bet!

File: utest.h [hunk 2]
```diff
@@ -1554,21 +1558,43 @@ int utest_main(int argc, const char *const argv[]) {
 
     ns = utest_ns();
     errno = 0;
+
+    pid_t pid = fork();
```
Comment (sheredom) [inline utest.h:1562]: We'll need to do some perf testing to see how much slower this is (I suspect it _might_ be).

File: utest.h (hunk 1 above)
Comment (Peyman-EsteghamatRad) [reply utest.h]: yes I think cross platform compatibility will be a problem. I will check if there is a solution for that.

**Comment by Peyman-EsteghamatRad** [comment]
Hey I tried a new approach to cover also Windows. I tested it on my own windows and linux machine and all the tests pass. Do you maybe know what the problem could be?

**Pull request closed without merging**